### PR TITLE
refactor(api): adding support for cloud download

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

**Comment by dustmop** [comment]
Is this ready for review? It's set as draft and failing CI

**Comment by Arqu** [comment]
No  not yet, sorry I tagged you. Need to sort out the tests first.

**Comment by dustmop** [comment]
No problem. Was just checking because https://github.com/qri-io/cloud/pull/207 is dependent on this, was confused.

**Comment by Arqu** [comment]
This is now ready for review. Bear in mind this is still pending on landing that migration work first (and serves as the base branch here for now)

File: api/dataset_test.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func TestDatasetHandlers(t *testing.T) {
 		{"GET", "/export/me/cities/at/map/QmPRjfgUFrH1GxBqujJ3sEvwV3gzHdux1j4g8SLyjbhwot", nil},
 		{"DELETE", "/", nil},
 	}
-	runHandlerTestCases(t, "export", h.ZipDatasetHandler, exportCases, true)
+	runHandlerTestCases(t, "export", h.ZipDatasetHandler("/export"), exportCases, true)
```
Comment (dustmop) [inline api/dataset_test.go:88]: This is fine for now, but fyi, we are working on removing `/export`, and will soon require callers to switch to get. See https://github.com/qri-io/qri/issues/1176. Hoping to remove export after 0.9.9 is released.

Comment (Arqu) [reply api/dataset_test.go:88]: Yep, once we start the move away from /export, plan is a) we continue to support the usecase b) happy to update to the new codepath, just that it's a bit too much work to unblock download just now.

**Pull request merged**